gtda/time_series/embedding.py

```diff
@@ -513,7 +444,238 @@ def resample(self, y, X=None):
+                 ensure_last_value=True):
+        self.time_delay = time_delay
+        self.dimension = dimension
+        self.stride = stride
+        self.flatten = flatten
+        self.ensure_last_value = ensure_last_value
+
+    def fit(self, X, y=None):
+        """Do nothing and return the estimator unchanged.
+
+        This method is here to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray or list
+            Input collection of time series. A 2D array or list of 1D arrays is
+            interpreted as a collection of univariate time series. A 3D array
+            or list of 2D arrays is interpreted as a collection of multivariate
+            time series, each with shape ``(n_variables, n_timestamps)``. More
+            generally, :math`N`-dimensional arrays or lists of
+            (:math`N-1`)-dimensional arrays (:math:`N \\geq 3`) are interpreted
+            as collections of tensor-valued time series, each with time indexed
+            by the last axis.
+
+        y : None
+            There is no need for a target, yet the pipeline API requires this
+            parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        check_time_series(X, copy=False)
+        validate_params(self.get_params(), self._hyperparameters)
+        self._is_fitted = True
+
+        return self
+
+    def transform(self, X, y=None):
+        """Compute the Takens embedding of each entry in `X`.
+
+        Parameters
+        ----------
+        X : ndarray or list
+            Input collection of time series. A 2D array or list of 1D arrays is
+            interpreted as a collection of univariate time series. A 3D array
+            or list of 2D arrays is interpreted as a collection of multivariate
+            time series, each with shape ``(n_variables, n_timestamps)``. More
+            generally, :math`N`-dimensional arrays or lists of
+            (:math`N-1`)-dimensional arrays (:math:`N \\geq 3`) are interpreted
+            as collections of tensor-valued time series, each with time indexed
+            by the last axis.
+
+        y : None
+            Ignored.
+
+        Returns
+        -------
+        Xt : ndarray or list
+            The result of performing a Takens embedding of each entry in `X`
+            with the given parameters. If `X` is a 2D array or a list of 1D
+            arrays, `Xt` is a 3D array or a list of 2D arrays (respectively),
+            each entry of which has shape ``(n_points, dimension)`` where
+            ``n_points = (n_timestamps - time_delay * (dimension - 1) - 1) // \
+            stride + 1``. If `X` is an :math`N`-dimensional array or a list of
+            (:math`N-1`)-dimensional arrays (:math:`N \\geq 3`), the output
+            shapes depend on the `flatten` parameter:
+
+                - if `flatten` is ``True``, `Xt` is still a 3D array or a
```

Comment:
I am still not a huge fan of this `flatten` thing. But anyways, let's agree to disagree xD

Reply:
Ok! Default is `True` anyway.